core/cloudflow-spark-testkit/src/main/scala/cloudflow/spark/testkit/SparkStreamletTestkit.scala

```diff
@@ -286,6 +281,9 @@ class QueryExecutionMonitor()(implicit ec: ExecutionContext) extends StreamingQu
     val id   = queryProgress.progress.id
     val rows = queryProgress.progress.numInputRows
     dataRows = dataRows + dataRows.get(id).map(v => id -> (v + rows)).getOrElse(id -> rows)
+    if (hasData & !dataAvailable.isCompleted) {
+      dataAvailable.success(dataRows.values.sum)
```

Comment:
you could use trySuccess if you are worried it will happen more than once. (success will throw exception if already completed)